Source/Core/Core/IOS/ES/Formats.cpp

```diff
@@ -596,20 +597,20 @@ std::vector<std::array<u8, 20>> SharedContentMap::GetHashes() const
 
 std::string SharedContentMap::AddSharedContent(const std::array<u8, 20>& sha1)
 {
-  auto filename = GetFilenameFromSHA1(sha1);
-  if (filename)
-    return *filename;
+  if (auto filename = GetFilenameFromSHA1(sha1))
+    return *std::move(filename);
```

Comment:
`filename` is a `std::optional<std::string>`, so here I am using the rvalue ref-qualified `operator*` overload to (if I am not mistaken) avoid a copy that RVO can't help avoid. Should I not use auto type deduction here so that is more clear?

Reply:
I wasn't aware that RVO doesn't help in this case. I'll defer to your judgment on this one.

Reply:
Here is a typical test demonstrating this fact: https://godbolt.org/z/G5saeaMbq.